docs/node-mixin/lib/prom-mixin.libsonnet

```diff
@@ -66,17 +66,12 @@ local table = grafana70.panel.table;
         datasource='$datasource',
         span=6,
         format='percentunit',
-        max=1,
         min=0,
         stack=true,
       )
       .addTarget(prometheus.target(
         |||
-          (
-            (1 - sum without (mode) (rate(node_cpu_seconds_total{%(nodeExporterSelector)s, mode=~"idle|iowait|steal", instance="$instance", %(clusterLabel)s="$cluster"}[$__rate_interval])))
-          / ignoring(cpu) group_left
-            count without (cpu, mode) (node_cpu_seconds_total{%(nodeExporterSelector)s, mode="idle", instance="$instance", %(clusterLabel)s="$cluster"})
-          )
+          (1 - sum without (mode) (rate(node_cpu_seconds_total{%(nodeExporterSelector)s, mode=~"idle|iowait|steal", instance="$instance", %(clusterLabel)s="$cluster"}[$__rate_interval])))
```

Comment:
I'm pretty sure we don't want to include `iowait` and `steal` here.

If we want this to be stacked CPU utilization, we probably want this:

```suggestion
          (avg without (mode) (1-rate(node_cpu_seconds_total{%(nodeExporterSelector)s, mode="idle", instance="$instance", %(clusterLabel)s="$cluster"}[$__rate_interval])))
```

Reply:
I vaguely remember a discussion about `steal`. But I don't remember the details. (But it feels like I have included `steal` for a reason.)

Why would you also toss `iowait`? Isn't the whole idea here to find out if you don't utilize your CPUs for whatever reason, including being stuck waiting for IO?

Reply:
`iowait` is not CPU time. It's accounted for in CPU metrics but it's actually phantom time when nothing is done. You can have both 100% idle and 100% `iowait` Leading to -100% CPU use in this calculation.

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/kernel/sched/cputime.c?id=HEAD#n222 looks more like the wait time is _either_ added to `idle` _or_ to `iowait`. But I'm just shooting in the dark here. Maybe that's not the relevant code, or it gets changed later on its way to node_exporter.